sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java

```diff
@@ -323,9 +323,9 @@
   String TIMEMACHINE_MODE_VERSION = "version";
   String TIMEMACHINE_MODE_DAYS = "days";
   String TIMEMACHINE_MODE_PREVIOUS_VERSION = "previous_version";
-  String TIMEMACHINE_DEFAULT_PERIOD_1 = TIMEMACHINE_MODE_PREVIOUS_ANALYSIS;
-  String TIMEMACHINE_DEFAULT_PERIOD_2 = "30";
-  String TIMEMACHINE_DEFAULT_PERIOD_3 = TIMEMACHINE_MODE_PREVIOUS_VERSION;
+  String TIMEMACHINE_DEFAULT_PERIOD_1 = TIMEMACHINE_MODE_PREVIOUS_VERSION;
+  String TIMEMACHINE_DEFAULT_PERIOD_2 = TIMEMACHINE_MODE_PREVIOUS_ANALYSIS;
+  String TIMEMACHINE_DEFAULT_PERIOD_3 = "30";
```

Comment:
Changing these constants do not work if plugins use a previous version of API. Indeed constant values are inlined by compiler. I suggest to move them from API to core.